src/controllers/objecting.entity.name.controller.ts

```diff
@@ -8,5 +11,16 @@ import { OBJECTIONS_COMPANY_NUMBER } from "../model/page.urls";
  * @param next
  */
 export const post = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
-  return res.redirect(OBJECTIONS_COMPANY_NUMBER)
+  const shareIdentityString: string = req.body.shareIdentity;
+  const fullNameValue: string = req.body.fullName;
+  const shareIdentityValue: boolean = (shareIdentityString === "yes") ? true :  false;
+  const objectionCreate: ObjectionCreate = { shareIdentity: shareIdentityValue, fullName: fullNameValue };
+
+  const session: Session | undefined  = req.session;
+  if (session) {
+    addObjectionCreateToObjectionSession(session, objectionCreate);
+    return res.redirect(OBJECTIONS_COMPANY_NUMBER);
+  } else {
+    throw new Error("Session not present");
```

Comment:
i'm not sure if the throwing of the error from the controller will result in the error page being shown.
next(e) will definitely show the error page.

Reply:
Replaced this with a next call containing the error.